app/src/main/java/com/jerboa/db/AppDB.kt

```diff
@@ -142,8 +149,30 @@ val APP_SETTINGS_DEFAULT = AppSettings(
     usePrivateTabs = false,
     secureWindow = false,
     blurNSFW = true,
+    saveSearchHistory = true,
+)
+
+@Entity
+data class SearchHistory(
+    @PrimaryKey @ColumnInfo(name = "text") val text: String,
+    @ColumnInfo(name = "timestamp") val timestamp: Long,
 )
 
+@Dao
+interface SearchHistoryDao {
+    @Query("SELECT * FROM SearchHistory")
+    fun history(): Flow<List<SearchHistory>>
```

Comment:
Use `LiveData` like the other examples. Don't code from scratch: use the conventions currently here.